packages/rum-recorder/src/domain/record/privacy.ts

```diff
@@ -36,30 +39,53 @@ export function nodeOrAncestorsShouldBeHidden(node: Node | null): boolean {
   return nodeOrAncestorsShouldBeHidden(node.parentNode)
 }
 
-// Returns true if the given DOM node should have it's input events
-// ignored. Ancestors are not checked.
-export function nodeShouldHaveInputIgnored(node: Node): boolean {
-  return (
-    isElement(node) &&
-    (node.getAttribute(PRIVACY_ATTR_NAME) === PRIVACY_ATTR_VALUE_INPUT_IGNORED ||
-      node.classList.contains(PRIVACY_CLASS_INPUT_IGNORED) ||
-      // if element is an HTMLInputElement, check the type is not to be ignored by default
-      (isInputElement(node) && PRIVACY_INPUT_TYPES_TO_IGNORE.includes(node.type)))
-  )
-}
+/**
+ * Returns the given node input privacy mode without checking the privacy mode on the node
+ * ancestors. Returns 'undefined' if the privacy mode cannot be determined and should be taken from
+ * an ancestor.
+ */
```

Comment:
FMU, it seems that we get the closest privacy mode to determine the privacy mode of the input.
so an input masked with an ignored parent would return masked.
Returning ignored would seem less surprising to me.
wdyt?

Reply:
That's debatable! IMO using the closest privacy mode is simpler to understand, instead of establishing a prority between them. Let's make a quick poll to get the thoughts of other devs.